### PR TITLE
Issue 958 flickr importer 3rd iter

### PR DESCRIPTION
**Pull request by alejandrodob**

### CODE REVIEW

File: pybossa/core.py [hunk 1]
```diff
@@ -283,6 +282,18 @@ def setup_social_networks(app):
         print inst
```
Comment (teleyinex) [inline pybossa/core.py]: Can you use the app.logger for loggin these issues?

Comment (alejandrodob) [inline pybossa/core.py]: yes. I will also leave the print as it is quite useful for local PyBossa installations, what do you think?

File: pybossa/importers.py [hunk 2]
```diff
@@ -160,13 +153,39 @@ def _get_album_info(self, album_id):
...
+            content['photo'] += rest_photos
+            return content
+        if type(res.tex) is dict:
+            error_message = json.loads(res.text).get('message')
+        else:
+            error_message = res.text
+        raise BulkImportException(error_message)
+
+    def _is_valid_response(self, response):
+        return (response.status_code == 200
```
Comment (teleyinex) [inline pybossa/importers.py]: Could this through an exception? In that case, it will fail, right?

**Comment by teleyinex** [comment]
:cake: after you answer those questions ;-)

File: pybossa/importers.py (hunk 2 above)
Comment (alejandrodob) [inline pybossa/importers.py]: you mean to change it like this?

    def _get_album_info(self, album_id):
        url = 'https://api.flickr.com/services/rest/'
        payload = {'method': 'flickr.photosets.getPhotos',
                   'api_key': self.api_key,
                   'photoset_id': album_id,
                   'format': 'json',
                   'nojsoncallback': '1'}
        res = requests.get(url, params=payload)
        if self._is_valid_response(res):
            content = json.loads(res.text)['photoset']
            total_pages = content.get('pages')
            rest_photos = self._remaining_photos(url, payload, total_pages)
            content['photo'] += rest_photos
            return content

    def _is_valid_response(self, response):
        if type(response.tex) is dict:
            error_message = json.loads(response.text).get('message')
        else:
            error_message = response.text
        valid = (response.status_code == 200
                    and json.loads(response.text).get('stat') == 'ok')
        if not valid:
            raise BulkImportException(error_message)
        return valid


Yes, this seems to work ok

Comment (teleyinex) [inline pybossa/importers.py]: That's perfect!

File: pybossa/core.py (hunk 1 above)
Comment (teleyinex) [inline pybossa/core.py]: yep, keep it and add the new solution so you can check in the log files what happened too.

**Comment by therealmarv** [comment]
:cake: from me

**Pull request merged**